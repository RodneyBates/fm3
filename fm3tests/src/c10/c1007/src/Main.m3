
(* -----------------------------------------------------------------------1- *)
(* This file is part of the FM3 Modula-3 compiler.                           *)
(* Copyright 2024        Rodney M. Bates.                                    *)
(* rodney.m.bates@acm.org                                                    *)
(* Licensed under the MIT License.                                           *)
(* -----------------------------------------------------------------------2- *)

MODULE Main

; TYPE T = INTEGER
; TYPE INTEGER = T (* Reserved id declared. *)
; TYPE V = REAL 
; TYPE T = CHAR    (* Duplicate declaration . *) 
; TYPE U = BOOLEAN

; TYPE
    T1 = INTEGER
  ; INTEGER = T (* Reserved id declared. *)
  ; V1 = REAL 
  ; T1 = CHAR    (* Duplicate declaration . *) 
  ; U1 = BOOLEAN

; BEGIN
  END Main 
.

