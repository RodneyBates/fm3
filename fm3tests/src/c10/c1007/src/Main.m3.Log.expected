[31mERROR: [0mMain.m3 (14,7) Identifier "INTEGER" is predefined and cannot be redeclared (2.8.2).
[31mERROR: [0mMain.m3 (21,4) Identifier "INTEGER" is predefined and cannot be redeclared (2.8.2).
[31mERROR: [0mMain.m3 (23,4) Duplicate declaration of "T1", ignored, original at (20,4).
[31mERROR: [0mMain.m3 (16,7) Duplicate declaration of "T", ignored, original at (13,7).
